python/ray/tune/tests/test_result_grid.py

```diff
@@ -344,6 +361,35 @@ def train_func(config):
     assert set(checkpoint_data) == {5, 6}
 
 
+def test_result_grid_cloud_path(ray_start_2_cpus, tmpdir):
+    # Test that checkpoints returned by ResultGrid point to URI
+    # if upload_dir is specified in SyncConfig.
+    sync_config = tune.SyncConfig(upload_dir="s3://bucket", syncer=MockSyncer())
+
+    def trainable(config):
+        for i in range(5):
+            checkpoint = Checkpoint.from_dict({"model": i})
+            session.report(metrics={"metric": i}, checkpoint=checkpoint)
+
+    tuner = tune.Tuner(
+        trainable,
+        run_config=air.RunConfig(
+            sync_config=sync_config,
+        ),
+        tune_config=tune.TuneConfig(
+            metric="metric",
+            mode="max",
+        ),
+    )
+    results = tuner.fit()
```

Comment:
We can delete the experiment directory here to simulate the checkpoint not being available on this node.